### PR TITLE
feat: add time filter and fix some bug for skipOnFail

### LINKED ISSUE #3915: [Bug][framework] skip on fail not work
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

image
My bp's skipOnFail is true.

image

Although the pipeline is triggered by this bp, its `skipOnFail` is false.

### What do you expect to happen

the pipeline's skipOnFail is true.

### How to reproduce

1. create a bp with skipOnFail=true
2. run it.

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by likyh**
### Summary
1. add time filter
2. fix some bugs for skipOnFail
3. change time format when pass to gitlab&github. It is no different, and just for good smell.

### Does this close any open issues?
Closes #3915

### Screenshots
image
image

### CODE REVIEW

File: config-ui/src/hooks/useDataScopesManager.jsx [hunk 1]
```diff
@@ -224,7 +224,7 @@ function useDataScopesManager({
 
   const getGithubProjects = useCallback(
     (c) =>
-      c.scope.map(
+      c.scopes.map(
```
Comment (likyh) [inline config-ui/src/hooks/useDataScopesManager.jsx:227]: Let the edit page can open in new blueprints. But the scope can not edit still.

File: config-ui/src/hooks/usePipelineManager.jsx [hunk 2]
```diff
@@ -179,41 +179,48 @@ function usePipelineManager(
...
-        console.log('>> RAW PIPELINES RUN DATA FROM API...', p.data?.pipelines)
-        let pipelines = p.data && p.data.pipelines ? [...p.data.pipelines] : []
-        // @todo: remove "ID" fallback key when no longer needed
-        pipelines = pipelines.map((p) => ({ ...p, ID: p.id }))
-        setPipelines(pipelines)
-        setPipelineCount(p.data ? p.data.count : 0)
-        setTimeout(() => {
-          setIsFetchingAll(false)
-        }, fetchTimeout)
+  const fetchAllPipelines = useCallback(
```
Comment (likyh) [inline config-ui/src/hooks/usePipelineManager.jsx:182]: fotmat by lint

File: config-ui/src/pages/blueprint/create/step-four/index.tsx [hunk 3]
```diff
@@ -108,12 +111,20 @@ export const StepFour = () => {
...
+          <p>Select the data range you wish to collect. DevLake will collect the last six months of data by default.</p>
+          <StartFromSelector
+            date={createdDateAfter}
+            onSave={onChangeCreatedDateAfter}
+          />
+      </div>
       <div className='block'>
         <h3>Frequency</h3>
         <p>Blueprints will run recurringly based on the sync frequency.</p>
+        <p style={{ margin: '10px 0' }}>{description}</p>
```
Comment (likyh) [inline config-ui/src/pages/blueprint/create/step-four/index.tsx:127]: https://www.figma.com/file/Eux4Dhorxp7M0TwzOnL6Yh/Config-UI-for-V1.0?node-id=2610%3A21361
pm say it should by left-right layout. But I don't know how to complete it, @mintsweet need your help.

File: plugins/github/tasks/cicd_run_collector.go [hunk 4]
```diff
@@ -58,7 +59,7 @@ func CollectRuns(taskCtx core.SubTaskContext) errors.Error {
 		Query: func(reqData *helper.RequestData) (url.Values, errors.Error) {
 			query := url.Values{}
 			if data.CreatedDateAfter != nil {
-				query.Set("created", fmt.Sprintf(">%s", data.CreatedDateAfter.String()))
+				query.Set("created", fmt.Sprintf(">%s", data.CreatedDateAfter.Format(time.RFC3339)))
```
Comment (likyh) [inline plugins/github/tasks/cicd_run_collector.go:62]: It's no effect. I promise.

File: services/blueprint_makeplan_v200.go [hunk 5]
```diff
@@ -72,7 +74,7 @@ func genPlanJsonV200(
 	sourcePlans := make([]core.PipelinePlan, len(connections))
 	scopes := make([]core.Scope, 0, len(connections))
 	for i, connection := range connections {
-		if len(connection.Scopes) == 0 {
+		if len(connection.Scopes) == 0 && connection.Plugin != `webhook` {
```
Comment (likyh) [inline services/blueprint_makeplan_v200.go:77]: for webhook's bp.

File: config-ui/package.json [hunk 6]
```diff
@@ -24,6 +25,7 @@
     "babel-plugin-module-resolver": "^4.1.0",
     "cron-parser": "^4.3.0",
     "cron-validate": "^1.4.3",
+    "date-fns": "^2.29.3",
```
Comment (mintsweet) [inline config-ui/package.json:28]: Why is another time library introduced? Can `dayjs` not meet the demand?

File: config-ui/src/components/blueprints/StartFromSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+  return (
+    <>
+      <div
+        className='start-from'
+      >
+        <div
+          className="123"
+          style={{display: 'flex', marginBottom: '10px'}}>
+          <Tag
```
Comment (mintsweet) [inline config-ui/src/components/blueprints/StartFromSelector.tsx:68]: Can use a config loop to survive?

File: config-ui/src/components/blueprints/StartFromSelector.tsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    onSave: (newDate: string | null, isUserChange?: boolean) => void,
+  }) => {
+
+  const formatDate = useCallback((date: Date) => format(date, "yyyy-MM-dd HH:mm"), []);
+  const parseDate = useCallback((str: string) => new Date(str), []);
+  const chooseDate = (date: Date) => {
+    onSave(format(date, "yyyy-MM-dd'T'HH:mm:ssxxx"))
+  }
+
+  const now = new Date()
```
Comment (mintsweet) [inline config-ui/src/components/blueprints/StartFromSelector.tsx:43]: useMemo returns these values?

File: config-ui/src/pages/blueprint/create/step-four/index.tsx (hunk 3 above)
Comment (mintsweet) [reply config-ui/src/pages/blueprint/create/step-four/index.tsx:127]: You can put it here first, I will modify it later

File: config-ui/package.json (hunk 6 above)
Comment (likyh) [reply config-ui/package.json]: fixed

File: config-ui/src/components/blueprints/StartFromSelector.tsx (hunk 7 above)
Comment (likyh) [reply config-ui/src/components/blueprints/StartFromSelector.tsx]: fixed

File: config-ui/src/components/blueprints/StartFromSelector.tsx (hunk 8 above)
Comment (likyh) [reply config-ui/src/components/blueprints/StartFromSelector.tsx]: fixed

**Review by mintsweet — APPROVED** [review]
LGTM.

**Review by klesh — DISMISSED** [review]
LGTM

**Review by klesh — APPROVED** [review]
LGTM

**Pull request merged**